homeassistant/core.py

```diff
@@ -820,31 +822,43 @@ async def async_stop(self, exit_code: int = 0, *, force: bool = False) -> None:
 
         # stage 1
         self.state = CoreState.stopping
```

Comment:
Why don't we introduce a new core state too? Also, shouldn't we set the state to `CoreState.stopping` just before firing `EVENT_HOMEASSISTANT_STOP` and not 20s before that?

Reply:
You're right, but the renaming of the concerned event was [a recent development](https://github.com/home-assistant/core/pull/91165/commits/27b6b6cbfec44881a98af9e426b58ca288450ee6#r1379937259), which made me forget that there was a `CoreState` named that way which could cause a confusion.